src/core/service.c

```diff
@@ -1481,6 +1481,7 @@ static Service *service_get_triggering_service(Service *s) {
 }
 
 static int service_spawn(
+                const char *reason,
```

Comment:
hmm, "reason", is that the right word? Maybe call it `caller`? Because that is what is passed here?